aten/src/ATen/native/cuda/CuFFTUtils.h

```diff
@@ -66,19 +70,261 @@ static inline void CUFFT_CHECK(cufftResult error)
   }
 }
 
-class CufftHandle {
-public:
-  explicit CufftHandle() {
-    CUFFT_CHECK(cufftCreate(&raw_plan));
+
+// This POD struct is used to let us easily compute hashes of the
+// parameters
+// It will be the **key** to the plan cache.
+struct CuFFTParams
+{
+  at::ScalarType scalar_type;
```

Comment:
Consider suffixing all of the members with a `_`? Will make it easier to distinguish parameter updates.